core/src/test/java/org/apache/calcite/test/TopDownOptTest.java

```diff
@@ -65,92 +70,184 @@
         + "group by mgr,deptno,comm\n"
         + "order by comm desc nulls last, deptno nulls first";
-    sql(sql).check();
+    // Always use sorted agg
+    Query.create(sql)
+        .removeRule(EnumerableRules.ENUMERABLE_AGGREGATE_RULE)
+        .addRule(EnumerableRules.ENUMERABLE_SORTED_AGGREGATE_RULE)
+        .check();
   }
 
   @Test void testSortMergeJoin() {
     final String sql = "select * from\n"
         + "sales.emp r join sales.bonus s on r.ename=s.ename and r.job=s.job\n"
         + "order by r.job desc nulls last, r.ename nulls first";
-    sql(sql).check();
+    Query.create(sql).removeRules(NON_DETERMINISTIC_JOIN_ORDER_RULES).check();
   }
 
   @Test void testSortMergeJoinRight() {
     final String sql = "select * from\n"
         + "sales.emp r join sales.bonus s on r.ename=s.ename and r.job=s.job\n"
         + "order by s.job desc nulls last, s.ename nulls first";
-    sql(sql).check();
+    Query.create(sql).removeRules(NON_DETERMINISTIC_JOIN_ORDER_RULES).check();
   }
 
   @Test void testMergeJoinDeriveLeft1() {
     final String sql = "select * from\n"
         + "(select ename, job, max(sal) from sales.emp group by ename, job) r\n"
         + "join sales.bonus s on r.job=s.job and r.ename=s.ename";
-    sql(sql).check();
+    Query.create(sql)
+        .removeRules(NON_DETERMINISTIC_JOIN_ORDER_RULES)
+        .removeRules(NON_MERGE_JOIN_RULES)
+        .removeRule(EnumerableRules.ENUMERABLE_AGGREGATE_RULE)
+        .addRule(EnumerableRules.ENUMERABLE_SORTED_AGGREGATE_RULE)
+        .check();
   }
 
   @Test void testMergeJoinDeriveLeft2() {
     final String sql = "select * from\n"
         + "(select ename, job, mgr, max(sal) from sales.emp group by ename, job, mgr) r\n"
         + "join sales.bonus s on r.job=s.job and r.ename=s.ename";
-    sql(sql).check();
+    Query.create(sql)
+        .removeRules(NON_DETERMINISTIC_JOIN_ORDER_RULES)
+        .removeRules(NON_MERGE_JOIN_RULES)
+        .removeRule(EnumerableRules.ENUMERABLE_AGGREGATE_RULE)
+        .addRule(EnumerableRules.ENUMERABLE_SORTED_AGGREGATE_RULE)
+        .check();
   }
 
   @Test void testMergeJoinDeriveRight1() {
     final String sql = "select * from sales.bonus s join\n"
         + "(select ename, job, max(sal) from sales.emp group by ename, job) r\n"
         + "on r.job=s.job and r.ename=s.ename";
-    sql(sql).check();
+    Query.create(sql)
+        .removeRules(NON_DETERMINISTIC_JOIN_ORDER_RULES)
+        .removeRules(NON_MERGE_JOIN_RULES)
+        .removeRule(EnumerableRules.ENUMERABLE_AGGREGATE_RULE)
+        .addRule(EnumerableRules.ENUMERABLE_SORTED_AGGREGATE_RULE)
+        .check();
   }
 
   @Test void testMergeJoinDeriveRight2() {
     final String sql = "select * from sales.bonus s join\n"
         + "(select ename, job, mgr, max(sal) from sales.emp group by ename, job, mgr) r\n"
         + "on r.job=s.job and r.ename=s.ename";
-    sql(sql).check();
+    Query.create(sql)
+        .removeRules(NON_DETERMINISTIC_JOIN_ORDER_RULES)
```

Comment:
I think they should be removed by default

Reply:
Agreed. Removed these rules for all tests now (as default).

If later any tests have a need to enable it, such tests can just add relavants rules.